### PR TITLE
check return value of functions that call BIO_new()

### PR DESCRIPTION
**Pull request by 0x9527-zhou**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: fuzz/client.c [hunk 1]
```diff
@@ -84,9 +92,12 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
             }
         }
     }
+ end:
     SSL_free(client);
     ERR_clear_error();
     SSL_CTX_free(ctx);
+    BIO_free(in);
+    BIO_free(out);
```
Comment (t8m) [inline fuzz/client.c:100]: This is wrong as the ownership of the bios is transferred to the SSL object with the SSL_set_bio() above. Just free the bios in the unable to create BIO `if`.

File: fuzz/client.c [hunk 2]
```diff
@@ -65,13 +65,21 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
 
     /* This only fuzzes the initial flow from the client so far. */
     ctx = SSL_CTX_new(SSLv23_method());
+    if(ctx == NULL)
+        goto end;
```
Comment (t8m) [inline fuzz/client.c:69]: You need to also initialize the client to NULL at the declaration.

File: fuzz/client.c [hunk 3]
```diff
@@ -65,13 +65,21 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
 
     /* This only fuzzes the initial flow from the client so far. */
     ctx = SSL_CTX_new(SSLv23_method());
+    if(ctx == NULL)
```
Comment (t8m) [inline fuzz/client.c:68]: Please always add space after `if`.

**Review by t8m — CHANGES_REQUESTED** [review]

File: fuzz/client.c (hunk 3 above)
Comment (0x9527-zhou) [reply fuzz/client.c:68]: Thanks a lot.
Should I resubmit?

Comment (t8m) [reply fuzz/client.c:68]: > Should I resubmit?

The best way is to fix the issues and commit with `git commit -a --fixup <commit-id-you're-fixing>`

This way the changes can be reviewed separately.

Comment (0x9527-zhou) [reply fuzz/client.c:68]: OK,thanks

**Comment by 0x9527-zhou** [comment]
Close for wrong code.

**Comment by t8m** [comment]
> Close for wrong code.

As I wrote above, it is unnecessary to close the PR and open a new one to adjust the changes. You can just add further fixup commits next time.

**Pull request closed without merging**